ors-engine/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java

```diff
@@ -74,10 +76,38 @@ public CoreLandmarkStorage(Directory dir, ORSGraphHopperStorage graph, RoutingCH
         this.core = (RoutingCHGraphImpl) core;
         this.landmarksFilter = lmConfig.getEdgeFilter();
         setMinimumNodes(Math.min(getBaseNodes() / 2, 10000));
+        this.coreNodeIdDA = dir.create(getLandmarksFileName() + lmConfig.getName() + "_core_node_ids");
+    }
+
+    public void createCoreNodeIdDA() {
+        int maxNode = GraphUtils.getBaseGraph(core).getNodes();
+
+        // initialize the DataAccess with enough space to store all core node ids
+        long maxBytes = maxNode * CORE_MAP_ROW_LENGTH;
+        coreNodeIdDA.create(maxBytes);
+        coreNodeIdDA.ensureCapacity(maxBytes);
+        for (long pointer = 0; pointer < maxBytes; pointer += CORE_MAP_ROW_LENGTH) {
+            coreNodeIdDA.setInt(pointer, 0);
+        }
+
+        int coreNodeLevel = maxNode;
+        int index = 0;
+        for (int i = 0; i < maxNode; i++) {
+            if (core.getLevel(i) < coreNodeLevel)
+                continue;
+            coreNodeIdDA.setInt(i * CORE_MAP_ROW_LENGTH, index);
+            index++;
+        }
```

Comment:
Are the two separate loops necessary? Maybe they could be combined into one, e.g. by zeroing `coreNodeIdDA` only for indices fulfilling the condition `core.getLevel(i) < coreNodeLevel)`.

Reply:
Addressed, please have a look